lib/matplotlib/rcsetup.pyi

```diff
@@ -156,3 +156,6 @@ def validate_hist_bins(
 ) -> Literal["auto", "sturges", "fd", "doane", "scott", "rice", "sqrt"] | int | list[
     float
 ]: ...
+
+def validate_legend_loc(loc: str | int | tuple[float]
```

Comment:
Should this be 
```suggestion
def validate_legend_loc(loc: str | int | tuple[float, float]
```

and the same on the return?

Reply:
Actually, validators in this file are currently type hinted as `Any` for the input, as the whole point is to pass unknown input and determine if they are valid for the expected types.

We use (and have for a long time) exceptions here. If we had instead used boolean-returning functions, we could potentially use [PEP 647 TypeGuards](https://peps.python.org/pep-0647/) for this idea, but with exceptions, this is pretty much how to do it (as exceptions are not type hinted)

But the point about it being a 2-tuple of float is valid.

Reply:
(With the asterisk on using PEP 647 being that it is python version 3.10+, but as bool is not how these work, that is a moot point anyway)